analyses/cell-type-neuroblastoma-04/exploratory-notebooks/scanvi-singler-comparison.Rmd

```diff
@@ -0,0 +1,682 @@
+# for this notebook, we want just a single Unknown
+recoded_celltype_pal["Unknown"] <- recoded_celltype_pal["unknown_singler"]
+
+nbatlas_palette_df <- readr::read_tsv(nbatlas_palette_file)
+nbatlas_celltype_pal <- nbatlas_palette_df$hex_color
+names(nbatlas_celltype_pal) <- nbatlas_palette_df$NBAtlas_label
+
+# sample metadata
+sample_metadata <- readr::read_tsv(sample_metadata_file)
+```
+
+
+## scANVI posterior probabilities
+
+Before comparing to `SingleR`, we should get a sense of reliability for the `scANVI` results.
+`scANVI` returns a cell-specific posterior probability for each label, so we'll begin by looking at these distributions.
+
+First, what is the distribution of posterior probabilities for assigned labels?
+We'll make a histogram below using a very visible color to ensure the plot is legible, where each panel is a library, and libraries are ordered by size.
+
+```{r fig.height=12, fig.width=15}
+scanvi_df |>
+  tidyr::separate(cell_id, into = c("library_id", "barcode")) |>
+  dplyr::add_count(library_id) |>
+  dplyr::mutate(
+    library_id = glue::glue("{library_id} (N={n})"),
+    library_id = factor(library_id), 
+    library_id = forcats::fct_infreq(library_id), 
+    library_id = forcats::fct_rev(library_id)
+  ) |>
+  ############ into ggplot ################
+  ggplot() + 
+  aes(x = scanvi_posterior) + 
+  # ugly but _very visible_ color
+  geom_histogram(bins = 100, fill = "magenta", color = "magenta") + 
+  facet_wrap(vars(library_id), scales = "free_y", ncol = 7)
+```
+
+
+Across libraries, the median posterior probability is always exceptionally high at which suggests `scANVI` confidence is roughly the same across libraries.
+
+Let's instead look at these probabilities on a per-cell type basis, this time with a ridge plot including a line for the median value for each distribution:
+
+```{r fig.height=8, fig.width=8}
+scanvi_ridgeplot <- scanvi_df |>
+  dplyr::group_by(scanvi) |>
+  dplyr::mutate(scanvi = glue::glue("{scanvi} (n={dplyr::n()})")) |>
+  ########### into ggplot ############
+  ggplot() + 
+    aes(y = forcats::fct_infreq(scanvi), x = scanvi_posterior) + 
+    # show the median
+    ggridges::stat_density_ridges(quantile_lines = TRUE, quantiles = 2, alpha = 0.7) +
+    labs(x = "posterior probability", y = "scANVI label (number of cells)") 
+scanvi_ridgeplot
+```
+
+While annotation reliability is about the same on a per-library basis, it seems that certain cell types may be much easier to classify than others which may be a factor of how well represented they are in the `NBAtlas` reference.
+There are very few cells in general associated with the cell types that have very low median posterior probabilities (`NKT cell`, `Migratory cDC`, `Circulating NK cell`).
+
+A `0.75` threshold seems like it may be a good middle-ground for preserving labels while removing low-confidence annotations:
+
+```{r fig.height=8, fig.width=8}
+scanvi_ridgeplot +
+  geom_vline(xintercept = 0.75, color = "firebrick")
```

Comment:
I think you can just do this in the first plot, I don't think we need two versions.